swarm.go

```diff
@@ -374,11 +374,21 @@ func (s *Swarm) Start(srv *p2p.Server) error {
 	log.Info("Updated bzz local addr", "oaddr", fmt.Sprintf("%x", newaddr.OAddr), "uaddr", fmt.Sprintf("%s", newaddr.UAddr))
 
 	if s.config.SwapEnabled {
-		err := s.swap.Deploy(context.Background(), s.backend, s.config.Path)
-		if err != nil {
-			return fmt.Errorf("Unable to deploy swap contract: %v", err)
+		if s.config.Contract != (common.Address{}) {
+			address := s.config.Contract
+			if err := cswap.ValidateCode(context.Background(), s.backend, address); err != nil {
+				return fmt.Errorf("Provided address not a chequebook smart-contract: %v", err)
```

Comment:
in case the `ValidateCode` call returns an error, there are 2 possibilities:
1. the operation of reading the code at this address can result in an error.
2. the byte code doesn't match—the resulting error string here would be: `Provided address not a chequebook smart-contract: not a swap contract`.

to avoid redundancy, i would suggest preceding the `err` text with something like `Error validating code at provided address:`

Reply:
Indeed scarily similar to the comment by @holisticode ... Resolved!